### PR TITLE
Switch to google-auth lib for authentication (bug 1864638)

### LINKED ISSUE #251: move away from oauth2client
The oauth2client lib is deprecated, its repo is archived, and says "We recommend you use [google-auth](https://google-auth.readthedocs.io/) and [oauthlib](http://oauthlib.readthedocs.io/). For more details on the deprecation, see [oauth2client deprecation](https://google-auth.readthedocs.io/en/latest/oauth2client-deprecation.html)."

### PR DESCRIPTION
**Pull request by jcristau**
This is not backwards-compatible: we now require the service account credentials in json format instead of pkcs12.

Fixes #251

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [7.0.0] - 2023-XX-XX
+
+### Changed
+* Removed support for pushing to the Amazon store.
+* Switched Google Play Store authentication to the google-auth library.  The credentials now need to be passed as a json file instead of an email address and PKCS#12 file.
```
Comment (JohanLorenzo) [inline CHANGELOG.md:9]: Nice!

File: README.md [hunk 2]
```diff
@@ -37,27 +37,17 @@ source venv/bin/activate
...
-1. Some errors might happen when executing `mozapkpublisher/push_apk.py`
-    1. You might have errors like
-        * Errors in from_p12_keyfile in oauth2client/service_account.py or
-        * ImportError: cannot import name `_openssl_crypt`
-            * `pip uninstall oauth2client`
-            * `pip install oauth2client==2.0.0`
-            * `pip install google-api-python-client==1.5.0`
-    1. Symbol not found: `_BIO_new_CMS`
-        * `pip uninstall cryptography`
-        * `LDFLAGS="-L/usr/local/opt/openssl/lib" pip install cryptography --no-use-wheel`
```
Comment (JohanLorenzo) [inline README.md:49]: Glad to see these lines gone

File: README.md [hunk 3]
```diff
@@ -37,27 +37,17 @@ source venv/bin/activate
...
 > A guide to manually publish APKs onto Google Play Store
 
-1. Generate a Google Play Store p12 certificate. This certificate needs to have write access to the app you want to publish. In this context, "app" means Fennec, Fennec Beta or Fennec Nightly.
+1. Generate a Google Play Store json certificate. This certificate needs to have write access to the app you want to publish. In this context, "app" means Fennec, Fennec Beta or Fennec Nightly.
 1. Execute the steps defined in the section above.
 1. Download the latest signed builds. For instance, for Fennec Nightly: `./mozapkpublisher/get_apk.py --latest-nightly`
 1. 
 ```sh
-./mozapkpublisher/push_apk.py --no-gp-string-update --track beta --credentials /path/to/your/googleplay/creds.p12 --service-account your-service-account@iam.gserviceaccount.com x86.apk arm.apk
+./mozapkpublisher/push_apk.py --no-gp-string-update --track beta --credentials /path/to/your/googleplay/creds.json x86.apk arm.apk
```
Comment (JohanLorenzo) [inline README.md:50]: Much cleaner!

File: mozapkpublisher/common/store.py [hunk 4]
```diff
@@ -189,23 +189,16 @@ def transaction(service_account, credentials_file_name, package_name, *, contact
...
-        # service_account_name, is the Email address created for the Service
-        # account. It must be the email address associated with the key that
-        # was created.
         scope = 'https://www.googleapis.com/auth/androidpublisher'
-        credentials = ServiceAccountCredentials.from_p12_keyfile(
-            service_account,
+        credentials = service_account.Credentials.from_service_account_file(
             credentials_file_name,
-            scopes=scope
+            scopes=[scope],
```
Comment (JohanLorenzo) [inline mozapkpublisher/common/store.py:197]: 👍

File: mozapkpublisher/test/test_push_apk.py [hunk 5]
```diff
@@ -24,9 +24,6 @@
 apk_arm = NamedTemporaryFile()
 
 APKS = [apk_x86, apk_arm]
-SERVICE_ACCOUNT = 'foo@developer.gserviceaccount.com'
-CLIENT_ID = 'client'
-CLIENT_SECRET = 'secret'
```
Comment (JohanLorenzo) [inline mozapkpublisher/test/test_push_apk.py:29]: Thanks for cleaning up these unused variable. I don't remember why they were here in the first place.

File: requirements/base.in [hunk 6]
```diff
@@ -1,9 +1,8 @@
 aiohttp
 androguard
 google-api-python-client
+google-auth
 humanize
 mozilla-version
-oauth2client
-pyOpenSSL   # Even not imported once in mozapkpublisher, it's needed to open p12 files
```
Comment (JohanLorenzo) [inline requirements/base.in:7]: I like this comment is not needed anymore.

**Review by JohanLorenzo — APPROVED** [review]
Looks great to me! I love the new authentication mechanism.

File: mozapkpublisher/common/utils.py [hunk 7]
```diff
@@ -47,14 +47,12 @@ def is_firefox_version_nightly(firefox_version):
...
-                        help='Google service account')
     parser.add_argument('--secret', required=True,
-                        help='File that contains google credentials')
+                        help='File that contains google credentials (json)')
     parser.add_argument('--do-not-contact-server', action='store_false', dest='contact_server',
                         help='''Prevent any request to reach the APK server. Use this option if
-you want to run the script without any valid credentials nor valid APKs. --service-account and
---credentials must still be provided (you can just fill them with random string and file).''')
+you want to run the script without any valid credentials nor valid APKs. --credentials must
+still be provided (you can just fill them with random string and file).''')
```
Comment (gbrownmozilla) [inline mozapkpublisher/common/utils.py:55]: nit: "with random file"

**Review by gbrownmozilla — APPROVED** [review]

File: mozapkpublisher/test/test_push_apk.py (hunk 5 above)
Comment (jcristau) [reply mozapkpublisher/test/test_push_apk.py:29]: They were part of the amazon code, and should have been removed in #253.

File: mozapkpublisher/common/utils.py (hunk 7 above)
Comment (jcristau) [reply mozapkpublisher/common/utils.py]: Thanks, fixed (as "you can pass a random file name").

**Pull request merged**